packages/core/src/internal/proc.js

```diff
@@ -289,7 +134,7 @@ export default function proc(env, iterator, parentContext, parentEffectId, meta,
       resolvePromise(effect, currCb)
     } else if (is.iterator(effect)) {
       // resolve iterator
-      proc(env, effect, taskContext, effectId, meta, /* isRoot */ false, currCb)
+      proc(env, effect, task.context, effectId, meta, /* isRoot */ false, currCb)
     } else if (effect && effect[IO]) {
       const effectRunner = effectRunnerMap[effect.type]
       effectRunner(env, effect.payload, currCb, executingContext)
```

Comment:
what about `effectRunner(env, task, effect.payload, currCb, digestEffect)`? 🤔 

The first argument is constant during the whole saga execution.
The second argument is constant during the executing of the task.
The third argument changes every time we yield a new effect.
The fourth is the callback, and the last is an argument we rarely use.